versions.tf

```diff
@@ -0,0 +1,3 @@
+terraform {
+  required_version = ">= 0.12.6"
```

Comment:
Good call. I've made the changes and updated the `README.md` to show the required AWS version.

Reply:
Hashicorp recommended practice for modules is to use `>=` 
https://www.terraform.io/docs/configuration/modules.html#provider-version-constraints-in-modules

Reply:
I saw that, but I can't really understand why they recommend it like that?

Terraform's `required_version` with `>= 0.12.6` will be too optimistic. I am quite certain this module will have to be updated with the future major releases of Terraform, same for AWS provider.

Let's merge now and see how things change in the future. Maybe I am wrong :)

@dpiddockcmp Do you agree?

Reply:
I can see the logic for Terraform itself, given the changes seen between 0.10, 0.11 and 0.12. But for the aws provider there wasn't that much difference between 1 and 2 for this module. 3 may well be the same and it does not block users from upgrading.

Reply:
The above documentation seems clear that the AWS provider should be `aws = ">= 2.23.0"`. I'll push up an update.

Reply:
[Terraform AWS Provider Version 3 Upgrade Guide](https://www.terraform.io/docs/providers/aws/guides/version-3-upgrade.html) exists, so there will be some breaking changes :)

Merging now and we will see how it goes.

Reply:
You were too fast, @eytanhanig . Let's be less optimistic and set it to "~> 2.23" as you had it earlier.

Reply:
Should I wait for confirmation from @dpiddockcmp so that we have consensus?

Reply:
I would say no need to wait because I want us to experience the versioning for real (not in theory).

If this is a wrong decision, users will report it during a few hours and we review it (I verified this multiple times).

PS: Thanks to people who don't pin module's version and use `master` branch :)

Reply:
@antonbabenko Done.